test/WebJobs.Script.Tests.Integration/Host/StandbyManager/StandbyManagerE2ETests_Windows.cs

```diff
@@ -79,6 +79,37 @@ public async Task ZipPackageFailure_DetectedOnSpecialization()
             host.Dispose();
         }
 
+
+        [Fact]
+        public async Task StandbyModeE2E_DotnetIsolated_WarmupSucceeds()
+        {
+            _settings.Add(EnvironmentSettingNames.AzureWebsiteInstanceId, Guid.NewGuid().ToString());
+
+            var environment = new TestEnvironment(_settings);
+            environment.SetEnvironmentVariable(RpcWorkerConstants.FunctionWorkerRuntimeSettingName, RpcWorkerConstants.DotNetIsolatedLanguageWorkerName);
+
+            await InitializeTestHostAsync("Windows", environment);
+
+            await VerifyWarmupSucceeds();
+
+            await TestHelpers.Await(() =>
+            {
+                // wait for the trace indicating that the host has started in placeholder mode.
+                var logs = _loggerProvider.GetAllLogMessages().Where(p => p.FormattedMessage != null).Select(p => p.FormattedMessage).ToArray();
+                return logs.Contains("Job host started") && logs.Contains("Host state changed from Initialized to Running.");
+            }, userMessageCallback: () => string.Join(Environment.NewLine, _loggerProvider.GetAllLogMessages().Select(p => $"[{p.Timestamp.ToString("HH:mm:ss.fff")}] {p.FormattedMessage}")));
+
+            var logLines = _loggerProvider.GetAllLogMessages().Where(p => p.FormattedMessage != null).Select(p => p.FormattedMessage).ToArray();
+
+            Assert.Single(logLines.Where(l => l.EndsWith("[FunctionsNetHost] Starting FunctionsNetHost")));
+            Assert.Equal(1, logLines.Count(p => p.Contains("Creating StandbyMode placeholder function directory")));
+            Assert.Equal(1, logLines.Count(p => p.Contains("StandbyMode placeholder function directory created")));
+            Assert.Equal(1, logLines.Count(p => p.Contains("Host is in standby mode")));
+
+            // Validating Functions worker runtime with function metadata does not happen in placeholder mode.
+            Assert.Equal(0, logLines.Count(p => p.Contains("The 'FUNCTIONS_WORKER_RUNTIME' is set to 'dotnet-isolated', which does not match the worker runtime metadata found in the deployed function app artifacts. The deployed artifacts are for 'CSharp'")));
```

Comment:
Might consider making this test more robust to error message changes by either just asserting that there are no Warnings logged, or verify no logs with the error code WorkerRuntimeDoesNotMatchWithFunctionMetadataErrorCode. Otherwise, if the error message is changed in the future, this test may incorrectly miss things.